### ISSUE TITLE
Postgres error - numeric column without precision and scale

### ISSUE THREAD

**Issue by josh-gree** [ibis-project/ibis#1273]
If a postgres table has a column with type `numeric` without a specified precision and scale then on trying to view the table in ibis;

```
t = con.table('table')
t
```

the following error is given;

```
TypeError: unsupported format string passed to NoneType.__format__
```

This comes from the `__repr__` method of `datatypes.Schema` object. Somewhere the precision and scale are parsed as `None` rather than some default value and the format fails...

**Comment by josh-gree** [comment]
Anybody got any thoughts on this? I think it's quite important to be able to support numeric types in postgres without a specified precision and scale!?

**Comment by cpcloud** [comment]
I should be able to take a look at this over the upcoming weekend.

**Comment by josh-gree** [comment]
Thanks! Would be much appreciated.

**Comment by ian-r-rose** [comment]
I also ran into this error recently. The postgres documentation for [numeric](https://www.postgresql.org/docs/9.0/datatype-numeric.html) states:

> without any precision or scale creates a column in which numeric values of any precision and scale can be stored, up to the implementation limit on precision. A column of this kind will not coerce input values to any particular scale, whereas numeric columns with a declared scale will coerce input values to that scale. (The SQL standard requires a default scale of 0, i.e., coercion to integer precision. We find this a bit useless. If you're concerned about portability, always specify the precision and scale explicitly.)

The SQLAlchemy numeric type seems to interpret both precision and scale as `None` when not specified. I'm unsure as to whether we should select defaults for these values in that case at the SQLAlchemy layer, or whether `dt.Decimal` should allow for `None` for those values.

**Comment by cpcloud** [comment]
Closed by #1969
